logger.go

```diff
@@ -14,11 +14,17 @@ import (
 var DefaultLogger = NewLogger(nil)
 
 // Log emits a log event to the default logger.
+//
+// Deprecated: Use the log/slog standard library package (or internally,
```

Comment:
Maybe clarify that "internally" means "if you're a Segment engineer, use this package instead" etc, since this is the public repo and an external engineer may not get the "internal" meaning